### PR TITLE
Added transcription APIs and events

### PR DESCRIPTION
**Pull request by vivekmore-msft**
Added following APIs and events to support live transcription.
- Add StartTranscription API 
- Add StopTranscription API 
- Add UpdateTranscriptionData(Change Language) API 
- Update answer Api to include transcription config 
- Update create call Api to include transcription config 
- Add models to support above changes

### CODE REVIEW

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs [hunk 1]
```diff
@@ -36,6 +36,11 @@ public AnswerCallOptions(string incomingCallContext, Uri callbackUri)
         /// </summary>
         public MediaStreamingOptions MediaStreamingOptions { get; set; }
 
+        /// <summary>
+        /// Live Transcription Configuration.
+        /// </summary>
+        public TranscriptionOptions TranscriptionOption { get; set; }
```
Comment (bharat-kalyan-namburi) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/AnswerCallOptions.cs:42]: TranscriptionOptions?

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CallConnectionProperties.cs [hunk 2]
```diff
@@ -19,7 +19,8 @@ public class CallConnectionProperties
             CommunicationIdentifier sourceIdentity,
             PhoneNumberIdentifier sourceCallerIdNumber,
             string sourceDisplayName,
-            string mediaSubscriptionId
+            string mediaSubscriptionId,
+            string dataSubscriptionId
```
Comment (bharat-kalyan-namburi) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/CallConnectionProperties.cs:23]: the word data might not be clear to the developers

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs [hunk 3]
```diff
@@ -40,6 +40,11 @@ public CreateCallOptions(CallInvite callInvite, Uri callbackUri)
         /// </summary>
         public MediaStreamingOptions MediaStreamingOptions { get; set; }
 
+        /// <summary>
+        /// Live Transcription Configuration.
+        /// </summary>
+        public TranscriptionOptions TranscriptionOption { get; set; }
```
Comment (bharat-kalyan-namburi) [inline sdk/communication/Azure.Communication.CallAutomation/src/Models/CreateCallOptions.cs:46]: same here, TranscriptionOptions

File: sdk/communication/Azure.Communication.CallAutomation/src/Models/CallConnectionProperties.cs (hunk 2 above)
Comment (vivekmore-msft) [reply sdk/communication/Azure.Communication.CallAutomation/src/Models/CallConnectionProperties.cs:23]: Will update it later

**Review by bharat-kalyan-namburi — APPROVED** [review]

**Pull request merged**